pkg/controllers/management/auth/project_cluster/cluster_handler.go

```diff
@@ -172,55 +176,64 @@ func (l *clusterLifecycle) createProject(name string, cond condition.Cond, obj r
 		if err != nil {
 			return obj, fmt.Errorf("error accessing project object %v: %w", obj, err)
 		}
+
+		clusterName := metaAccessor.GetName()
+
 		// Attempt to use the cache first
-		projects, err := l.projectLister.List(metaAccessor.GetName(), labels.AsSelector())
+		projects, err := l.projectLister.List(clusterName, labels.AsSelector())
 		if err != nil || len(projects) > 0 {
 			return obj, err
 		}
 
 		// Cache failed, try the API
-		projects2, err := l.projects.List(metaAccessor.GetName(), metav1.ListOptions{LabelSelector: labels.String()})
+		projects2, err := l.projects.List(clusterName, metav1.ListOptions{LabelSelector: labels.String()})
 		if err != nil || len(projects2.Items) > 0 {
 			return obj, err
 		}
 
-		annotation := map[string]string{}
+		clusterAnnotations := metaAccessor.GetAnnotations()
+		annotations := map[string]string{}
 
-		creatorID := metaAccessor.GetAnnotations()[CreatorIDAnnotation]
-		if creatorID != "" {
-			annotation[CreatorIDAnnotation] = creatorID
+		if creatorID := clusterAnnotations[CreatorIDAnnotation]; creatorID != "" {
+			annotations[CreatorIDAnnotation] = creatorID
+		}
+
+		if creatorPrincipalName := clusterAnnotations[creatorPrincipalNameAnnotation]; creatorPrincipalName != "" {
+			annotations[creatorPrincipalNameAnnotation] = creatorPrincipalName
 		}
 
 		if name == project.System {
 			latestSystemVersion, err := systemimage.GetSystemImageVersion()
 			if err != nil {
 				return obj, err
 			}
-			annotation[project.SystemImageVersionAnnotation] = latestSystemVersion
+			annotations[project.SystemImageVersionAnnotation] = latestSystemVersion
 		}
 
 		project := &apisv3.Project{
 			ObjectMeta: metav1.ObjectMeta{
 				GenerateName: "p-",
-				Annotations:  annotation,
+				Annotations:  annotations,
 				Labels:       labels,
+				Namespace:    clusterName,
 			},
 			Spec: apisv3.ProjectSpec{
 				DisplayName: name,
-				Description: fmt.Sprintf("%s project created for the cluster", name),
-				ClusterName: metaAccessor.GetName(),
+				Description: name + " project created for the cluster",
+				ClusterName: clusterName,
 			},
 		}
 		updated, err := l.addRTAnnotation(project, "project")
 		if err != nil {
 			return obj, err
 		}
+
 		project = updated.(*apisv3.Project)
-		logrus.Infof("[%v] Creating %s project for cluster %v", ClusterCreateController, name, metaAccessor.GetName())
-		if _, err = l.mgr.Management.Projects(metaAccessor.GetName()).Create(project); err != nil {
-			return obj, err
-		}
-		return obj, nil
+
+		logrus.Infof("[%s] Creating %s project for cluster %s", ClusterCreateController, name, clusterName)
+		_, err = l.projects.Create(project)
+
+		return obj, err
```

Comment:
We were not doing this, but should we return the just created Project? Is the passed `obj` a Project?

Reply:
We should not. The `obj` is cluster.